presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayAllocator.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.spi.block;
+
+import com.facebook.presto.spi.api.Experimental;
+
+/**
+ * Manages the creation and return of primitive arrays, to be used within an operator to avoid repeated array allocation.
```

Comment:
Could you also mention that the caller is expected to borrow arrays of increasing size?

Reply:
Now that the concrete implementation is package-private in the `operator` package in presto-main, I don't think this feedback applies, because that may not be the intended usage.

Reply:
@tdcmeehan The logic of throwing away smaller arrays still exists. Would it make sense to document that `the caller is expected to borrow arrays of increasing size` in the implementation class?